File: php/examples/payroll/php/administrator.php
<?php
include_once('deductions.php');
    class administrator extends person{
    use deductions;
        
        public function __construct(){
            $daysWorked=12;
            $dayValue=87000;
            $salary=$daysWorked*$dayValue;
            $transport= ($salary<2000000)?200000:0;
        
        parent:: __construct("Carla Sofia","Zuleta Dussan",$daysWorked,$dayValue,$transport,$salary,"Administrator"); 
        $this-> initDeductions($salary);
        }
            public function calculateSalary():float{
                 return ($this->getDaysWorked()*$this->getDayValue())-$this->totalDeductions;
            }
            public function show():void{
                echo "<h1>Information of the worker: ".$this->getPosition()."</h1>";
                echo "<h2>Position: ".$this->getPosition(). "</h2>";
                echo "<p>Names and last names: " . $this->getNames() ." ". $this->getLastNames() ."</p>";
                echo "<p>Days worked: "  .$this->getDaysWorked(). "</p>";
                echo "<p>Day value: " .$this->getDayValue()."$"."</p>";
                echo "<p>With a salary of: ". $this->getSalary()."$"."</p>";
                echo "<p>With a auxiliary of transport of: ". $this->getTransport()."$". "</p>";
                $this->showDeductions();
                echo "<p>For obtain a fullpayment of: ". $this->calculateSalary()."$"."</p>";
            }
        }
?>

File: java/payroll/src/main/java/com/maicol/payroll/Administrator.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.maicol.payroll;

/**
 *
 * @author Maicol
 */
   public class Administrator extends person {

    public Administrator() {
        super("Carla Sofia", "Zuleta Dussan", 12, 87000, (12 * 87000 < 2000000) ? 200000 : 0, 12 * 87000, "Administrator");
    }

    @Override
    public double calculateSalary() {
        return (getDaysWorked() * getDayValue()) - 0; 
    }

    @Override
    public void show() {
        System.out.println("Information of the worker: " + getPosition());
        System.out.println("Position: " + getPosition());
        System.out.println("Names and last names: " + getNames() + " " + getLastNames());
        System.out.println("Days worked: " + getDaysWorked());
        System.out.println("Day value: " + getDayValue() + "$");
        System.out.println("With a salary of: " + getSalary() + "$");
        System.out.println("With a transport allowance of: " + getTransport() + "$");
        System.out.println("For obtaining a full payment of: " + calculateSalary() + "$");
    }
}
